src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java

```diff
@@ -1,42 +1,354 @@
+
+    @Autowired
+    private Producer producer;
+
+    @Autowired
+    RequestInterceptor requestInterceptor;
+    Logger log = LogManager.getLogger(WorkflowServiceImplV2.class);
+
+
+    /**
+     * Change the status of workflow application
+     *
+     * @param userToken
+     * @param wfRequest
+     * @return
+     */
+    @Override
+    public Response workflowTransition(String userToken, WfRequest wfRequest) {
+        Response response = new Response();
+        String changedStatus = null;
+        String userId = validateAuthTokenAndFetchUserId(userToken);
+        if (!CollectionUtils.isEmpty(wfRequest.getUpdateFieldValues())) {
+            List<HashMap<String, Object>> validTerms = new ArrayList<>();
+            ArrayList<String> invalidTerms = new ArrayList<>();
+            for (HashMap<String, Object> updatedField : wfRequest.getUpdateFieldValues()) {
+                if (StringUtils.isEmpty((String) updatedField.get(Constants.APPROVAL_STATUS)) || updatedField.get(Constants.APPROVAL_STATUS).equals(Constants.DRAFT)) {
+                    validTerms.add(updatedField);
+                } else {
+                    invalidTerms.add((String) updatedField.get(Constants.IDENTIFIER));
+                }
+            }
+            if (CollectionUtils.isNotEmpty(invalidTerms)) {
+                throw new InvalidDataInputException(Constants.TERM_APPROVAL_STATUS_ERROR + ": " + invalidTerms);
+            }
+            wfRequest.setUpdateFieldValues(validTerms);
+        }
+        HashMap<String, String> changeStatusResponse = changeStatus(wfRequest, userId);
+        changedStatus = changeStatusResponse.get(Constants.STATUS);
+        HashMap<String, Object> data = new HashMap<>();
+        data.put(Constants.STATUS, changedStatus);
+        data.put(Constants.WF_ID_CONSTANT, changeStatusResponse.get(Constants.WF_ID_CONSTANT));
+        response.put(Constants.MESSAGE, Constants.STATUS_CHANGE_MESSAGE + changedStatus);
+        response.put(Constants.DATA, data);
+        response.put(Constants.STATUS, HttpStatus.OK);
+        return response;
+    }
+
+    /**
+     * Get the application based on wfId
+     *
+     * @param userToken
+     * @param wfId
+     * @return Wf Application based on wfId
+     */
+    @Override
+    public Response getWfApplication(String userToken, String wfId) {
+        String userId = validateAuthTokenAndFetchUserId(userToken);
+        WfStatusEntityV2 applicationStatus = wfStatusRepoV2.findByUserIdAndWfId(userId, wfId);
+        List<Map<String, Object>> updateFieldValuesList = null;
+        if (StringUtils.isNotEmpty(applicationStatus.getUpdateFieldValues())) {
+            try {
+                updateFieldValuesList = mapper.readValue(applicationStatus.getUpdateFieldValues(), new TypeReference<List<Map<String, Object>>>() {
+                });
+            } catch (Exception ex) {
+                throw new ApplicationException(Constants.UPDATED_FIELD_VALUES_ERROR_MESSAGE, ex);
```

Comment:
Do not throw exception - update the response object with error code.

Reply:
updated